### PR TITLE
Update oauth2client.go

### PR DESCRIPTION
**Pull request by wora**
Mostly documentation changes.

### CODE REVIEW

File: go/src/oauth2client/oauth2client.go [hunk 1]
```diff
@@ -213,8 +207,7 @@ func (c ThreeLeggedClient) GetToken(scope string) (*Token, error) {
 	return verifyFlow(c.secret, scope, code)
 }
 
-// Run the two-legged oauth flow, will create a JWT token and use the JWT token
-// to fetch an oauth token. Returns the token Object
+// Run 2LO flow. Create a JWT token and use it to fetch an OAuth token.
```
Comment (bochunz) [inline go/src/oauth2client/oauth2client.go]: The Go tradition for comment for a public function starts with the function name.
e.g. GetToken runs 2LO flow. ...

File: go/src/oauth2client/oauth2client.go [hunk 2]
```diff
@@ -178,15 +173,15 @@ func createJWT(secret map[string]interface{}, scope string, pkey pkeyInterface)
 	return segments + "." + base64Encode(signedBytes), nil
 }
 
-// Interface for OAuth2 Client
+// Interface for OAuth 2 Client
```
Comment (bochunz) [inline go/src/oauth2client/oauth2client.go]: Client interface for OAuth 2.
(Don't forget a period at the end)

File: go/src/oauth2client/oauth2client.go (hunk 1 above)
Comment (wora) [inline go/src/oauth2client/oauth2client.go]: Will fix later.

File: go/src/oauth2client/oauth2client.go (hunk 2 above)
Comment (wora) [inline go/src/oauth2client/oauth2client.go]: Done.

**Comment by wora** [comment]
PTAL.

**Comment by bochunz** [comment]
LGTM

**Pull request merged**